actions/push-to-gcs/action.yaml

```diff
@@ -0,0 +1,95 @@
+name: Push to cloud storage
+description: Composite action to push to Google Cloud Storage
+inputs:
+  bucket:
+    description: |
+      Name of bucket to upload to. Will default to grafanalabs-${repository.id}-${environment}
+    default: ""
+  path:
+    description: |
+      Path to the object(s) to upload. Can either include a filename to upload 1 file,
+      or can be used in conjunction with the `glob` option to upload files from a path.
+      Valid examples include `thing.txt` and `path/to/thing.txt`.
+      Valid examples when also using `glob` include `path/to`.
+    required: true
+  bucket_path:
+    description: |
+      The path in the bucket to save the object(s). Defaults to the root of the bucket.
+    default: ""
+  environment:
+    description: |
+      Environment for uploading objects (can be either dev or prod).
+    default: dev
+  glob:
+    description: |
+      Glob pattern.
+    default: ""
+
+outputs:
+  uploaded:
+    description: "The list of successfully uploaded files."
+    value: ${{ steps.upload-files.outputs.uploaded }}
+
+runs:
+  using: composite
+  steps:
+    - name: Resolve GCP project
+      id: resolve-project
+      shell: bash
+      run: |
+        if [ "${{ inputs.environment }}" = "dev" ]; then
+          PROJECT="grafanalabs-dev"
+        elif [ "${{ inputs.environment }}" == "prod" ]; then
+          PROJECT="grafanalabs-global"
+        else
+          echo "Invalid environment. Valid environment variable inputs: dev, prod"
+          exit 1
+        fi
+        echo "project=${PROJECT}" >> ${GITHUB_OUTPUT}
+    - name: Construct destination
+      id: construct-destination
+      shell: bash
+      run: |
+        # Construct bucket name
+        if [ "${{ inputs.bucket }}" == "" ]; then
+          BUCKET="grafanalabs-${{ github.repository_id }}-${{ inputs.environment }}"
+        else
+          BUCKET="${{ inputs.bucket }}"
+        fi
+
+        # If glob is set, then we'll use that instead of object
```

Comment:
you mean instead of the path?

Reply:
Ah, good catch. I meant `object`, but I renamed `object` to `path` when I introduced the `glob` option. I will fix.